### PR TITLE
go trough the updater when uploading over webdav

### LINKED ISSUE #18006: Folder size propagation not triggered by Webdav upload
### Steps to reproduce

1. Create a folder "test"
1. Using a Webdav client (I used cadaver), upload a file "pic.jpg" into "test"
1. In the web UI, check the size of the folder "test"

### Expected result
Folder size is updated to reflect the size of the picture inside.

### Actual result
Folder size is still zero.

I noticed that in the database the propagation isn't happening after upload.
However the folder size will update itself later through other mechanisms, possibly through PROPFINDs or the scanner that the web UI triggers (ajax/scan.php) or the background cron scan.

This also means that the quota value (the total used space) will not be updated directly.

Uploading through the web UI properly propagates the sizes, so it's specific to Webdav.

### Versions
stable8.1 (a9524e250a1f53a20a68acd2449e90fa2d13ef51)

I will bisect to find out when this regression was introduced.
Not sure how critical. Setting to 8.1.1 now but 8.1.2 might be fine.

@DeepDiver1975 @icewind1991 @schiesbn

### PR DESCRIPTION
**Pull request by icewind1991**
Fixes #18006

cc @DeepDiver1975 @PVince81

### CODE REVIEW

**Comment by karlitschek** [comment]
nice catch :+1:

**Comment by PVince81** [comment]
Thanks. Can you add a unit test for this ?

File: lib/private/connector/sabre/file.php [hunk 1]
```diff
@@ -208,10 +208,9 @@ public function put($data) {
 			}
 
 			// since we skipped the view we need to scan and emit the hooks ourselves
-			$partStorage->getScanner()->scanFile($internalPath);
+			$this->fileView->getUpdater()->update($this->path);
```
Comment (schiesbn) [inline lib/private/connector/sabre/file.php]: The comment above should probably go from L210 to L212. By the way, what's the different between the local ````$view```` and ````$this->fileView```` ?

Comment (icewind1991) [inline lib/private/connector/sabre/file.php]: `$view` is the view for the user used for hooks, `$this->fileView` is the view exposed for the dav endpoint

**Comment by PVince81** [comment]
This files:scan was the entry point of another issue @schiesbn and I investigated last Friday: https://github.com/owncloud/core/issues/17594

I retested this case with this PR and the encrypted file is still uploaded properly, so it seems to work.

@icewind1991 did you think of possible side effects and can confirm that the change from scan() to update() will not have any ?

**Comment by icewind1991** [comment]
> did you think of possible side effects and can confirm that the change from scan() to update() will not have any ?

I dont think there will be any negative side effects

**Comment by PVince81** [comment]
Tested, works :+1:

**Comment by PVince81** [comment]
Backport this for 8.1.2 ? @karlitschek

**Comment by karlitschek** [comment]
please backport :+1:

**Comment by icewind1991** [comment]
8.1: 18108

**Pull request merged**